package.json

```diff
@@ -34,6 +34,7 @@
     "karma-coverage": "~2.0.3",
     "karma-jasmine": "~4.0.0",
     "karma-jasmine-html-reporter": "^1.5.0",
+    "ng-mocks": "^12.3.1",
```

Comment:
Spent some time mocking my component, not hard but looks awful and re-usable components are sure to have duplicates.

This works well assuming we're good with Shallow copies, to keep tests strictly to the unit and ignoring child behaviour excluding what it tells the parent.